### ISSUE TITLE
[World Cup Feature][Milestone 2][Accessibility - VoiceOver] - The More Options button on the widget is no longer selected after dismissing the context menu

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#33938]
**Steps to Reproduce**

1. Launch Firefox on the homepage.
2. Enable VoiceOver via iOS settings.
3. Navigate to the "Countdown to the World Cup" widget.
4. Tap the More Options (3-dot) button to open the context menu.
5. Dismiss the context menu.
6. Observe the focus state of the More Options button.

**Expected Behavior**

After dismissing the context menu, VoiceOver focus returns to the More Options button that triggered it.

**Actual Behavior**

After dismissing the context menu, the More Options button is no longer selected, and VoiceOver focus is lost as the Firefox Logo (and text) is selected

**Device & Build Information**

- Device: iPhone 16 (26.4)
- Build version: v9000 (72186)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-15867)
